### PR TITLE
Add checksum validation at the end of download

### PR DESCRIPTION
**Pull request by matszczygiel**

### CODE REVIEW

File: drop-core/src/status.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ pub enum Status {
     AuthenticationFailed = 30,
     StorageError = 31,
     DbLost = 32,
+    FileChecsumMismatch = 33,
```
Comment (Lipt0nas) [inline drop-core/src/status.rs:26]:
```suggestion
    FileChecksumMismatch = 33,
```

File: drop-transfer/src/error.rs [hunk 2]
```diff
@@ -46,6 +46,8 @@ pub enum Error {
     AuthenticationFailed,
     #[error("Storage error")]
     StorageError,
+    #[error("Checksum validation failed")]
+    CheckSumMismatch,
```
Comment (Lipt0nas) [inline drop-transfer/src/error.rs:50]:
```suggestion
    ChecksumMismatch,
```

File: drop-transfer/src/ws/server/v4.rs [hunk 3]
```diff
@@ -651,6 +656,20 @@ impl handler::Downloader for Downloader {
...
+            let file = std::fs::File::open(&path.0)?;
+            let csum = file::checksum(&mut io::BufReader::new(file))?;
+            crate::Result::Ok(csum)
+        })?;
+
+        if self.full_csum.get().await == csum {
+            Ok(())
+        } else {
+            Err(crate::Error::CheckSumMismatch)
+        }
```
Comment (Lipt0nas) [inline drop-transfer/src/ws/server/v4.rs:671]: Mega minor: To fall in-line with the refactor in `ws/server/mod.rs`, but can ignore this one

```suggestion
        if self.full_csum.get().await != csum {
            return Err(crate::Error::CheckSumMismatch)
        }
        
        Ok(())
```

File: drop-core/src/status.rs (hunk 1 above)
Comment (matszczygiel) [reply drop-core/src/status.rs]: Applied

File: drop-transfer/src/ws/server/v4.rs (hunk 3 above)
Comment (matszczygiel) [reply drop-transfer/src/ws/server/v4.rs]: Applied

**Review by Lipt0nas — APPROVED** [review]

**Pull request merged**